src/generation.rs

```diff
@@ -110,22 +121,78 @@ impl<'a> SerializeConfig<'a> {
     fn container_encoding_lookup(&self, prefix: &str, encoding_fields: &Vec<EncodingField>, var: &str) -> String {
         let encoding_lookup = match &self.encoding_var_in_option_struct {
             Some(namespace) => format!(
-                "{}.as_ref().and_then(|encs| encs.{}_{}_encodings.get({})).map(|e| e.clone())",
+                "{}.as_ref().and_then(|encs| encs.{}_{}_encodings.get({})).cloned()",
```

Comment:
what was the issue with clippy related to that?

Reply:
clippy doesn't like you using `.map|e| e.clone())` when `.cloned()` exists. It's in the style category.